docroot/modules/custom/mass_entity_usage/mass_entity_usage.module

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/**
+ * @file
+ * Contains mass_entity_usage.module.
+ */
+
+use Drupal\content_moderation\Entity\ContentModerationState;
+use Drupal\content_moderation\Entity\ContentModerationStateInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\mayflower\Helper;
+
+/**
+ * Implements hook_entity_usage_block_tracking().
+ */
+function mass_entity_usage_entity_usage_block_tracking($target_id, $target_type, $source_id, $source_type, $source_langcode, $source_vid, $method, $field_name, $count) {
+  // Block tracking of entities with a node source that is unpublished or trash.
+  if (!_mass_entity_usage_tracking_check($source_type, $source_id, $source_vid)) {
+    return TRUE;
+  }
+  return FALSE;
+}
+
+/**
+ * Find an id/revision_id record in the default field data table.
+ *
+ * @param string $entity_type
+ *   The entity type.
+ * @param int $id
+ *   The entity id.
+ * @param int $vid
+ *   The entity revision id.
+ *
+ * @return bool
+ *   If true, the content should be tracked. If false, do not track.
+ */
+function _mass_entity_usage_tracking_check($entity_type, $id, $vid) {
+  if ($entity_type == 'paragraph') {
+    // Verify the paragraph revision is in the paragraphs_item_field_data table.
+    $query = \Drupal::database()->select('paragraphs_item_field_data', 'pifd');
+    $query->fields('pifd', ['id', 'revision_id']);
+    $query->condition('id', $id);
+    $query->condition('revision_id', $vid);
+    $results = $query->execute()->fetchAll();
+    if (empty($results)) {
+      return FALSE;
+    }
+    // Load the paragraph revision.
+    $paragraph = \Drupal::service('entity_type.manager')->getStorage('paragraph')
+      ->loadByProperties(['id' => $id, 'revision_id' => $vid]);
+    // Get the parent node of the paragraph.
+    $parent_node = Helper::getParentNode(current($paragraph));
+    if (is_null($parent_node)) {
+      return FALSE;
+    }
+    // Check if the parent node should be tracked.
+    _mass_entity_usage_tracking_check('node', $parent_node->id(), $parent_node->getRevisionId());
+  }
+  if ($entity_type == 'node') {
+    // Load the node.
+    $node = \Drupal::entityTypeManager()->getStorage('node')->loadRevision($vid);
+    // Get the moderation state.
+    $content_moderation_state = ContentModerationState::loadFromModeratedEntity($node);
+    if ($content_moderation_state instanceof ContentModerationStateInterface
+      && !$content_moderation_state->get('moderation_state')->isEmpty()) {
+      $state_name = $content_moderation_state->get('moderation_state')->value;
+      // These states should be not be tracked.
+      $skipped_states = [
+        'unpublished',
+        'trash',
+      ];
```

Comment:
1. We have constants for each state so you dont have to use strings. See MassModeration class
1. It may make sense to make a method like  getPrepublishedStates in that same class for all the states we want here. Do this if you think we might need same grouping in the future.